src/actions/index.js

```diff
@@ -41,3 +49,17 @@ export const addDictionaryEntry = (entry) => {
     entry,
   };
 };
+
+export function batchAdd(charges) {
+  return (dispatch, state) => {
+    const oldCharges = state().get('credit', List());
+    let id = oldCharges.size ? oldCharges.last().get('id') + 1 : 1;
+
+    const dict = state().get('dictionary');
+    charges.map( (charge) => {
```

Comment:
also think that if you're using map, it generates a new array but looks like your idea here is actually mutate all elements inside your array, maybe forEach is better choice?